records_mover/records/targets/table/target.py

```diff
@@ -39,16 +39,20 @@ def __init__(self,
                  schema_name: str,
                  table_name: str,
                  db_engine: Engine,
-                 db_driver: Callable[[Union[Engine, Connection]], DBDriver],
+                 db_driver: Callable[[Optional[Union['Engine', 'Connection']],
+                                      Optional[Connection],
+                                      Optional[Engine]], DBDriver],
```

Comment:
its just such an ugly type declaration, can we use a typedef or alias or something to describe this factory parameter that isnt really a db_driver but returns one?

Reply:
it's so annoying. I tried to create a Protocol for this but it didn't make it a lot clearer. I'm open to ideas but this is definitely a method and not a property so it has to be wrapped in this Callable type.

Reply:
Alright, I tried again to write a protocol for this function but we run into a lot of problems with imports and such; I think I'm going to have to say that, while this is extremely ugly, it at least works so I'm going to have to leave as is. </3